tests/unit/test_adapter_yaml.py

```diff
@@ -0,0 +1,567 @@
+              - item
+              - nested_dict:
+                  key: value
+            list3: [item, item]
+            """
+        )
+        config = YamlfixConfig()
+        config.flow_style_sequence = True
+
+        result = fix_code(source, config)
+
+        assert result == fixed_source
+
+    def test_sequence_block_style_enforcement_for_lists_longer_than_line_length(
+        self,
+    ) -> None:
+        """Fall back to multi-line list style 'block-style' if list would be longer than\
+            line_length and multiline flow-style is not enabled, even if flow-style\
+                is selected."""
+        source = dedent(
+            """\
+            looooooooooooooooooooooooooooooooooooongKey:
+              - item
+            list:
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+            list2:
+              - item
+              - item
+              - item
+            """
+        )
+        fixed_source = dedent(
+            """\
+            ---
+            looooooooooooooooooooooooooooooooooooongKey:
+              - item
+            list:
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+            list2: [item, item, item]
+            """
+        )
+        config = YamlfixConfig()
+        config.line_length = 40
+        config.flow_style_sequence = True
+
+        result = fix_code(source, config)
+
+        assert result == fixed_source
+
+    def test_sequence_flow_style_multiline_config(self) -> None:
+        """Make inline list style 'flow-style' configurable even for lists, that would\
+            wrap into the next line."""
+        source = dedent(
+            """\
+            looooooooooooooooooooooooooooooooooooongKey:
+              - item
+            list:
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+              - loooooooooooooooooooongItem
+            list2:
+              - item
+              - item
+              - item
+            """
+        )
+        fixed_source = dedent(
+            """\
+            ---
+            looooooooooooooooooooooooooooooooooooongKey: [
+              item]
+            list: [loooooooooooooooooooongItem, loooooooooooooooooooongItem,
+              loooooooooooooooooooongItem, loooooooooooooooooooongItem,
+              loooooooooooooooooooongItem, loooooooooooooooooooongItem]
+            list2: [item, item, item]
```

Comment:
I don't think this way of writing long lists is good. I'd remove this functionality. In what cases do you think it's interesting?